src/app/modules/myinfo/__tests__/myinfo.service.spec.ts

```diff
@@ -435,4 +380,192 @@ describe('MyInfoService', () => {
+      const result = await myInfoService.fetchMyInfoData(
+        MOCK_MYINFO_FORM as IPopulatedForm,
+        { MyInfoCookie: MOCK_SUCCESSFUL_COOKIE },
+      )
+
+      // Assert
+      expect(mockGetPerson).toHaveBeenCalledWith(
+        MOCK_ACCESS_TOKEN,
+        MOCK_REQUESTED_ATTRS.concat('uinfin' as MyInfoAttribute),
+        MOCK_ESRVC_ID,
+      )
+      expect(result._unsafeUnwrap()).toEqual(new MyInfoData(mockReturnedParams))
+    })
+
+    it('should throw MyInfoFetchError when getPerson fails once', async () => {
+      // Arrange
+      mockGetPerson.mockRejectedValueOnce(new Error())
+
+      // Act
+      const result = await myInfoService.fetchMyInfoData(
+        MOCK_MYINFO_FORM as IPopulatedForm,
+        { MyInfoCookie: MOCK_SUCCESSFUL_COOKIE },
+      )
+
+      // Assert
+      expect(mockGetPerson).toHaveBeenCalledWith(
+        MOCK_ACCESS_TOKEN,
+        MOCK_REQUESTED_ATTRS.concat('uinfin' as MyInfoAttribute),
+        MOCK_ESRVC_ID,
+      )
+      expect(result._unsafeUnwrapErr()).toEqual(new MyInfoFetchError())
+    })
+
+    it('should throw MyInfoCircuitBreakerError when getPerson fails 5 times', async () => {
+      // Arrange
+      mockGetPerson.mockRejectedValue(new Error())
+      for (let i = 0; i < 5; i++) {
+        await myInfoService.fetchMyInfoData(
+          MOCK_MYINFO_FORM as IPopulatedForm,
+          { MyInfoCookie: MOCK_SUCCESSFUL_COOKIE },
+        )
+      }
+
+      // Act
+      const result = await myInfoService.fetchMyInfoData(
+        MOCK_MYINFO_FORM as IPopulatedForm,
+        { MyInfoCookie: MOCK_SUCCESSFUL_COOKIE },
+      )
+
+      // Assert
+      // Last function call doesn't count as breaker is open, so expect 5 calls
+      expect(mockGetPerson).toHaveBeenCalledTimes(5)
+      expect(result._unsafeUnwrapErr()).toEqual(new MyInfoCircuitBreakerError())
+    })
+    it('should not validate the form if the cookie does not exist', async () => {
+      // Arrange
+      const expected = new MyInfoMissingAccessTokenError()
+
+      // Act
+      const result = await myInfoService.fetchMyInfoData(
+        MOCK_MYINFO_FORM as IPopulatedForm,
+        {},
+      )
+
+      // Assert
+      expect(result._unsafeUnwrapErr()).toEqual(expected)
+    })
+  })
+
+  describe('createFormWithMyInfo', () => {
+    // NOTE: Mocks the underlying circuit breaker implementation to avoid network calls
+    beforeEach(() => {
+      myInfoService = new MyInfoService(MOCK_SERVICE_PARAMS)
+    })
+
+    it('should return the filled form when the form and cookies are valid', async () => {
+      // Arrange
+      const MOCK_MYINFO_DATA = ({
+        getUinFin: jest.fn().mockReturnValue(MOCK_UINFIN),
+      } as unknown) as MyInfoData
+
+      const expected = {
+        prefilledFields: [],
+        spcpSession: { userName: MOCK_UINFIN },
+      }
+
+      // Spies to ensure that submethods have been called
+      const prefillMyInfoSpy = jest
+        .spyOn(myInfoService, 'prefillMyInfoFields')
```

Comment:
if this turns into a private method, you may have to combine the tests for `prefillMyInfoFields` and `createFormWithMyInfo`